### PR TITLE
Support dimension given as object.

### PR DESCRIPTION
**Pull request by gesundkrank**
Sorry. I'll try it again. Would still be nice to get some feedback on my PRs. 😉

### CODE REVIEW

**Comment by estliberitas** [comment]
I'm sorry. Quite busy these days. Will review in 12hrs.

File: lib/fields/dimension.js [hunk 1]
```diff
@@ -17,26 +17,30 @@ module.exports = dimension
...
  * @returns {object}
  */
-function dimension(dimension, outputName, fn) {
-  if (!dimension) {
+function dimension(dimensionOrObject, outputName, fn) {
+  if (!dimensionOrObject) {
     throw new FieldError('At least dimension must be specified')
   }
 
+  if (arguments.length === 1 && typeof  dimensionOrObject === 'object') {
```
Comment (estliberitas) [inline lib/fields/dimension.js]: `s/typeof  /typeof /`

File: lib/fields/dimension.js [hunk 2]
```diff
@@ -17,26 +17,30 @@ module.exports = dimension
  * In other cases DefaultDimensionSpec is created.
  *
  * @see http://druid.io/docs/0.6.121/DimensionSpecs.html
- * @param {string} dimension Dimension name
```
Comment (estliberitas) [inline lib/fields/dimension.js]: you can leave dimension name, changing just jsdoc

**Comment by estliberitas** [comment]
Could you update API doc too for dimension method?

**Comment by estliberitas** [comment]
LGTM except minor nits and question.

**Comment by gesundkrank** [comment]
Thx for the review. I submitted fixups and also updated the API documentation.

**Comment by estliberitas** [comment]
LGTM, releasing then.

File: README.md [hunk 3]
```diff
@@ -702,15 +702,17 @@ __Query#dataSource('query', subQuery)__
 
 Set DimensionSpec. 
 
-Depending on arguments length creates default or extraction dimension spec.
+Give dimension spec as first argument.
```
Comment (estliberitas) [inline README.md]: Change it to `If first argument is an object, then just use it as DimensionSpec.`?

Comment (gesundkrank) [inline README.md]: Done.

**Comment by gesundkrank** [comment]
Updated API docs, squashed and rebased.

**Pull request merged**